napalm_nxos_ssh/nxos_ssh.py

```diff
@@ -793,11 +793,10 @@ def get_facts(self):
                 _, domain_name = re.split(r".*Default domain.*is ", line)
                 domain_name = domain_name.strip()
                 break
-        if domain_name != 'Unknown' and hostname != 'Unknown':
-            if hostname.count(".") >= 2:
-                fqdn = hostname
-            else:
-                fqdn = '{}.{}'.format(hostname, domain_name)
+        if hostname.count(".") >= 2:
+            fqdn = hostname
+        elif domain_name != 'Unknown':
```

Comment:
I personally find this strange to have domain name as "Unknown" - wouldn't be more natural that if you didn't configure it, napalm to return you empty string? i.e., currently if you don't have a domain name configured it will tell you that your domain name is "unknown" which is very strange.

Reply:
Will fix...I think we need to start defining the unspecified values in napalm-base. We are getting too much variance because of this (i.e. everyone is guessing what the unspecified values should be).

Slightly different issue, but I somewhat think this definition of fqdn is wrong:
https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/base.py#L220

To me fqdn should be a null string if it isn't a fully-qualified domain-name (i.e. not just set to the hostname).

Reply:
Yeah, agreed on the defaults that we should make obvious.

I think should move this discussion elsewhere, as I have a different opinion on the FQDN: for me, that looks exactly as I would expect: no domain name, means FQDN is just the hostname.  And I see this on several platforms, including Arista EOS:

```bash
host#sh hostname | json
{
    "fqdn": "host",
    "hostname": "host"
}
```